query-engine/core/src/query_graph_builder/extractors/query_arguments.rs

```diff
@@ -24,7 +24,10 @@ pub fn extract_query_args(
                 }),
 
                 args::TAKE => Ok(QueryArguments {
-                    take: arg.value.try_into()?,
+                    take: {
+                        let n: Option<i64> = arg.value.try_into()?;
+                        n.into()
```

Comment:
```suggestion
                        i64::try_from(arg.value)?.into()
```

Reply:
Nope, it would not work because of error handling:

```
`?` could not convert error type `Infallible` to `QueryGraphBuilderError` [E0277]
 ```

Reply:
my bad, it needs to be
```suggestion
                        <Option<i64>>::try_from(arg.value)?.into()
```